### PR TITLE
[INLONG-5496][SDK] Refactor single topic manager in Sort SDK

### LINKED ISSUE #5496: [Improve][SDK] Refactor single topic manager in Sort SDK
### Description

differ the old single topic policy from the undergoing multi-topic policy #5495

### InLong Component

InLong SDK

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #5496

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]
LGTM

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+import org.apache.inlong.sdk.sort.util.StringUtil;
+import org.apache.inlong.tubemq.client.config.TubeClientConfig;
+import org.apache.inlong.tubemq.client.factory.MessageSessionFactory;
+import org.apache.inlong.tubemq.client.factory.TubeSingleSessionFactory;
+import org.apache.pulsar.client.api.AuthenticationFactory;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InLongSingleTopicManager extends InLongTopicManager {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java:52]: 1. Please add some Java docs for new classes.
2. Suggested changing the `InLongXxx` to `InlongXxx`.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java [hunk 2]
```diff
@@ -311,6 +328,13 @@ public enum ConsumeStrategy {
         lastest_absolutely
     }
 
+    public enum TopicManagerType {
+        // single topic manager
+        single_topic,
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java:333]: The enum fields should use uppercase letters.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+
+    private final Logger logger = LoggerFactory.getLogger(InLongSingleTopicManager.class);
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java:54]: This field should be static, and the name of `static final` should use uppercase letters.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java:52]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java (hunk 3 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InLongSingleTopicManager.java:54]: fixed

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java:333]: fixed, thx

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ClientContext.class})
+public class InlongSingleTopicManagerTest {
+
+    private InLongTopic inLongTopic;
+    private ClientContext clientContext;
+    private QueryConsumeConfig queryConsumeConfig;
+    private InlongTopicManager inLongTopicManager;
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java:53]: `inLongTopicManager` -> `inlongTopicManager`.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongTopicManagerFactory.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.inlong.sdk.sort.api.QueryConsumeConfig;
+import org.apache.inlong.sdk.sort.api.SortClientConfig.TopicManagerType;
+
+/**
+ * Inlong topic manager factory.
+ * To create single or multi topic fetcher manager according to the {@link TopicManagerType}
+ */
+public class InlongTopicManagerFactory {
+
+    public static InlongTopicManager createInLongTopicManager(
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongTopicManagerFactory.java:32]: `inLongTopicManager` -> `inlongTopicManager`.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/SortClientImpl.java [hunk 6]
```diff
@@ -39,7 +40,7 @@ public class SortClientImpl extends SortClient {
 
     private final ClientContext context;
 
-    private final InLongTopicManager inLongTopicManager;
+    private final InlongTopicManager inLongTopicManager;
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/SortClientImpl.java:43]: `inLongTopicManager` -> `inlongTopicManager`.

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        ConcurrentHashMap<String, InLongTopicFetcher> fetchers = new ConcurrentHashMap<>();
+        InLongTopicFetcher inLongTopicFetcherRmMock = PowerMockito.mock(InLongTopicFetcher.class);
+        fetchers.put(inLongTopic.getTopicKey(), inLongTopicFetcherRmMock);
+        Whitebox.setInternalState(inLongTopicManager, "fetchers", fetchers);
+        allFetchers = inLongTopicManager.getAllFetchers();
+        Assert.assertEquals(1, allFetchers.size());
+    }
+
+    @Test
+    public void testOfflineAllTp() {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java:145]: What does the `Tp` mean? 
If it is `topic`, then using the `Topics` is better.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManager.java [hunk 8]
```diff
@@ -0,0 +1,495 @@
...
+                if (preValue != null) {
+                    result = preValue;
+                    if (inLongTopicFetcher != null) {
+                        inLongTopicFetcher.close();
+                    }
+                    LOGGER.info("addFetcher create same fetcher {}", inLongTopic);
+                } else {
+                    result = inLongTopicFetcher;
+                    if (result != null
+                            && !initFetcher(result, inLongTopic)) {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManager.java:117]: Those two lines can be merged into one line.

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java (hunk 7 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java:145]: it means topic and partition

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java (hunk 4 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManagerTest.java:53]: there are too many naming problem, I will make a new issue to fix these.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongTopicManagerFactory.java (hunk 5 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongTopicManagerFactory.java:32]: there are too many naming problem, I will make a new issue to fix these.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManager.java (hunk 8 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongSingleTopicManager.java:117]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/SortClientImpl.java (hunk 6 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/SortClientImpl.java:43]: there are too many naming problem, I will make a new issue to fix these.

**Review by healchow — APPROVED** [review]

**Review by doleyzi — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**